src/components/titlePane/createTitlePane.ts

```diff
@@ -0,0 +1,88 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+import { DNode, Widget, WidgetFactory, WidgetProperties } from '@dojo/widget-core/interfaces';
+import themeable, { ThemeableMixin } from '@dojo/widget-core/mixins/themeable';
+
+import * as css from './styles/titlePane.css';
+import * as animations from '../../styles/animations.css';
+
+export interface TitlePaneProperties extends WidgetProperties {
+	collapsed?: boolean;
+	collapsible?: boolean;
+	enterAnimation?: string;
+	exitAnimation?: string;
+	title: string;
+
+	onRequestCollapse?(): void;
+	onRequestExpand?(): void;
+};
+
+export type TitlePane = Widget<TitlePaneProperties> & ThemeableMixin & {
+	onClickTitle?(): void;
+};
+
+export interface TitlePaneFactory extends WidgetFactory<TitlePane, TitlePaneProperties> { };
+
+const createTitlePane: TitlePaneFactory = createWidgetBase
+.mixin(themeable)
+.mixin({
+	mixin: {
+		baseClasses: css,
+
+		onClickTitle: function (this: TitlePane) {
```

Comment:
Should this be named `onTitleClick` instead? Just preference, but it would be good to stay consistent with other components. Also, all of these functions can be written without the `function` keyword:

```ts
...

onTitleClick(this: TitlePane) {
    // ...
}

...
```